src/transformers/training_args.py

```diff
@@ -148,7 +148,8 @@ class TrainingArguments:
         no_cuda (:obj:`bool`, `optional`, defaults to :obj:`False`):
             Whether to not use CUDA even when it is available or not.
         seed (:obj:`int`, `optional`, defaults to 42):
-            Random seed for initialization.
+            Random seed for initialization of the model's weights. To ensure reproducibility across runs, use the
+            :func:`~transformers.Trainer.model_init` function to instantiate the model.
```

Comment:
That's not correct. The seed is set and will affect the model creation if a `model_init` is passed, but it will also affect every bit of randomness (for instance the random samplers for shuffling the training set).
```suggestion
            Random seed that will be set at the beginning of training. To ensure reproducibility across runs, use the
            :func:`~transformers.Trainer.model_init` function to instantiate the model if it has some randomly initialized 
            parameters.
```